File: signup.py
from flask import Flask, request, jsonify, render_template
import mysql.connector
from werkzeug.security import generate_password_hash

app = Flask(__name__)

# Database connection
db = mysql.connector.connect(
    host="localhost",
    user="root",           # Replace with your MySQL username
    password="merijaan",   # Replace with your MySQL password
    database="siksha_mitra"   # Replace with your database name
)

cursor = db.cursor()

# Home route for testing
@app.route("/")
def home():
    return render_template("signup.html")  # Ensure the signup HTML is in a `templates` folder.

# API endpoint for sign-up
@app.route("/signup", methods=["POST"])
def signup():
    try:
        # Get data from the request
        data = request.json
        username = data.get("username") # type: ignore
        email_mobile = data.get("email_mobile") # type: ignore
        password = data.get("password") # type: ignore
        confirm_password = data.get("confirm_password") # type: ignore

        # Basic validation
        if not all([username, email_mobile, password, confirm_password]):
            return jsonify({"error": "All fields are required"}), 400

        if password != confirm_password:
            return jsonify({"error": "Passwords do not match"}), 400

        # Check if the user already exists
        cursor.execute("SELECT * FROM signup WHERE email_mobile = %s", (email_mobile,))
        if cursor.fetchone():
            return jsonify({"error": "Email/Mobile already registered"}), 400

        # Hash the password for security
        password_hash = generate_password_hash(password)

        # Insert user into the database
        cursor.execute(
            "INSERT INTO signup (username, email_mobile, password_hash) VALUES (%s, %s, %s)",
            (username, email_mobile, password_hash)
        )
        db.commit()

        return jsonify({"message": "User registered successfully"}), 201
    except Exception as e:
        print(f"Error: {e}")
        return jsonify({"error": "Something went wrong"}), 500

if __name__ == "__main__":
    app.run(debug=True)

File: signup.js
document.addEventListener("DOMContentLoaded", () => {
    const signupForm = document.getElementById("signup");
    const feedback = document.getElementById("feedback");

    signupForm.addEventListener("submit", (e) => {
        e.preventDefault();

        const username = document.getElementById("username").value.trim();
        const email = document.getElementById("email").value.trim();
        const password = document.getElementById("password").value;
        const confirmPassword = document.getElementById("confirmPassword").value;

        // Validation
        if (!username || !email || !password || !confirmPassword) {
            feedback.textContent = "All fields are required.";
            return;
        }
        if (password !== confirmPassword) {
            feedback.textContent = "Passwords do not match.";
            return;
        }
        if (password.length < 6) {
            feedback.textContent = "Password must be at least 6 characters.";
            return;
        }
        // Success message
        feedback.textContent = "Sign-up successful!";
        feedback.style.color = "green";

        // Redirect after success
        setTimeout(() => {
            window.location.href = "loginpage.html"; // Update as needed
        }, 2000);
    });
});

document.getElementById("signup").addEventListener("submit", async (e) => {
    e.preventDefault();

    const username = document.getElementById("username").value.trim();
    const emailMobile = document.getElementById("email/mobile_no.").value.trim();
    const password = document.getElementById("password").value;
    const confirmPassword = document.getElementById("confirmPassword").value;

    const response = await fetch("/signup", {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify({
            username,
            email_mobile: emailMobile,
            password,
            confirm_password: confirmPassword
        })
    });

    const result = await response.json();
    const feedback = document.getElementById("feedback");

    if (response.ok) {
        feedback.style.color = "green";
        feedback.textContent = result.message;
    } else {
        feedback.style.color = "red";
        feedback.textContent = result.error;
    }
});
